### DISCUSSION TITLE
Custom Format Compatibility

### DISCUSSION THREAD

**Discussion by dexterslab2013** [keirf/flashfloppy#434] [Q&A]
Hello!

I am currently restoring some vintage computer hardware, a quantel dpb-7001 digital paintbox, it was quite an iconic machine launched in 1981 but sold in very small numbers. I believe the one i own is the only one remaining, i have been working with a number of engineers from the company that made it to try and get this one running in it's 40th anniversary year

It originally used an 8" DSDD Shugart interface floppy drive so i need to replace it with FF/Gotek. I have done all the wiring and connected gotek and it works ok reading but with some errors. But writing does not work at all, the host reports disk errors and the data is totally corrupted.

The disk format is rather unusual, the host uses the drive as a library of tracks as it were, the disk is NOT formatted into sectors.

I am in contact with the person responsible for designing the format back in 1981 and i have done a good deal of research myself using scopes and logic analysers to determine the format.

My current understanding is the following:
Total Tracks: 65
RPMs: 360
Sides: 2
Data Per Track: 19,200 bytes excluding padding/sync
Encoding: MFM

Track layout looks like this (IP Being the Index Pulse)
```
| IP | Padding (0x00) | Sync Byte (0x53) | Data | Padding (0x00) | IP | ..... etc
```
i have only just started using gotek/FF so need a little help working this out as i get familiar with the finer points of disk emulators! As i said before i have this working but i think it just needs tweaking!

any help would be appreciated!!
mark

**Comment by keirf** [comment]
Are the disks hard sectored (multiple index holes / pulses per revolution)?

**Reply by dexterslab2013** [reply]
hi keirf!
No, they are soft-sectored disks

**Reply by keirf** [reply]
Oh yes I see there is no division into sectors. Well this should be doable with HFE image file. There is a mk_hfe.py script shipped with FlashFloppy. Try `python3 mk_hfe.py --rate=500 --rpm=360 myimage.hfe` and this really should work. Try another USB stick too.

**Reply by dexterslab2013** [reply]
Thanks i just tried the image that the script generated, my machine reports it wrote ok, but it's corrupted on read back.

I can however initialise the disk image on the machine (a bit like formatting)

Problem here is it's very hard for me to verify if it wrote the disk correctly and there is some other fault or if it wrote the disk badly!

**Reply by keirf** [reply]
Upload the initialised/written image an I can take a look.

**Reply by dexterslab2013** [reply]
Thank you!

I have attached two files, the ZIP contains the HFE images:
DPB_TestPicture.hfe: hfe generated with the FF script, formatted and test picture written to/read back from.
130505_BrushMaster.hfe: hfe image generated from Kryoflux raw files of an original disk that is required to boot the machine. Included for reference.

The brush master disk image usually reads ok but i do get errors when reading sometimes using gotek. Using the actual physical floppy it came from seems to work perfectly.

The other file is a video capture of me loading the image back in after saving, the test picture is colour bars with a grey circle in the centre.

Any insight into what was written to the test image would be of help

https://user-images.githubusercontent.com/33674152/109286094-bf10d900-7819-11eb-842b-6418c0c0bb05.mp4

[DPB_HFE_Files.zip](https://github.com/keirf/FlashFloppy/files/6048824/DPB_HFE_Files.zip)

**Reply by keirf** [reply]
Nothing too surprising in the HFE image file really. It looks like MFM. Formatting doesn't overrun the track. 

Maybe fixing reading first makes sense. A few options to try in FF.CFG:
```
index-suppression = no
track-change = realtime
```
Switch index-suppression back to yes if it doesn't help.

**Reply by keirf** [reply]
I recommend doing this using your known good disk (BrushMaster) by the way. If it basically works with random errors I think we can assume the image is okay too. So that gives a basis for exploring the link between the Gotek and the host.

**Reply by dexterslab2013** [reply]
thanks, i don't want to risk using the real brush master disk at the moment, it's probably the only one in the world. The other issue is there isn't enough space on the brush master to save an image.

Can we extract out the actual MFM decoded data from the HFE files? Seeing the entire track data written in hex might be useful to me

**Reply by keirf** [reply]
You don't need to touch the BrushMaster original disk, you have an image. Take a copy of that and work on why it's getting read errors, is my recommendation.

I will see about pulling you out hex track dumps.

**Reply by keirf** [reply]
Here are raw binary dumps of tracks, file per track. You will want to view using a hex viewer.
[rawmfm.zip](https://github.com/keirf/FlashFloppy/files/6049404/rawmfm.zip)

Bear in mind these are not byte aligned. Sync word if any can start at any bit offset.

**Reply by dexterslab2013** [reply]
Thanks i will take a look!

I have connected my logic analyser up the floppy interface and the gotek, see if i can track down any issues.

I have noticed something, if you take a look at the trace below, it's a read of track 2, side 1 of the BrushMaster disk. The B3 MFM DATA trace is the output of the gotek. There's periods of no data . Can you tell me if that comes from my hfe file or why they might be there?

![image](https://user-images.githubusercontent.com/33674152/109309132-cc8a8b00-783a-11eb-94e2-beb0129ec937.png)

Zoomed version:
![image](https://user-images.githubusercontent.com/33674152/109310040-f42e2300-783b-11eb-9d41-6a595ae3570c.png)
